File: clases 25-11-2025/Tipos de clases js/clase normal con interfaz/Clases/Libro.js
import ILibro from '../interfaces/ILibro.js';

export default class Libro extends ILibro {
    constructor(titulo) {
        super();
        this.titulo = titulo;
        this.autor = "Desconocido";
    }

    mostrarLibro() {
        console.log(`Título: ${this.titulo}, Autor: ${this.autor}`);
    }
}

File: clases 25-11-2025/tipos de clases Java/clase normal con interfaz/Libro.java
public class Libro implements ILibro {
    public String titulo;
    public String autor;

    // Constructor sobrecargado
    public Libro(String titulo) {
        this.titulo = titulo;
        this.autor = "Desconocido";
    }

    @Override
    public void mostrarLibro() {
        System.out.println("Título: " + titulo + ", Autor: " + autor);
    }
}
